lib/plausible/ingestion/write_buffer.ex

```diff
@@ -43,13 +44,13 @@ defmodule Plausible.Ingestion.WriteBuffer do
   @impl true
   def handle_cast({:insert, row_binary}, state) do
     state = %{state | buffer: [state.buffer | row_binary], buffer_size: state.buffer_size + 1}
+    ms_since_last_insert = now() - state.last_insert_at
 
-    if state.buffer_size >= state.max_buffer_size do
+    if state.buffer_size >= state.max_buffer_size and ms_since_last_insert >= :timer.seconds(1) do
```

Comment:
`:timer.seconds(1)` can be made configurable.

Reply:
In this implementation it's possible to keep the "larger than max" buffer for up to `flush_interval_ms`, until the timer fires.